apps/call.go

```diff
@@ -33,12 +33,19 @@ const (
 // TODO: what if a call needs a token and it was not provided? Return a call to
 // itself with Expand.
 type Call struct {
-	Path       string                 `json:"path,omitempty"`
-	Type       CallType               `json:"type"`
-	Values     map[string]interface{} `json:"values,omitempty"`
-	Context    *Context               `json:"context,omitempty"`
-	RawCommand string                 `json:"raw_command,omitempty"`
-	Expand     *Expand                `json:"expand,omitempty"`
+	Path   string      `json:"path,omitempty"`
+	Expand *Expand     `json:"expand,omitempty"`
+	State  interface{} `json:"state,omitempty"`
+}
+
+type CallRequest struct {
+	*Call
```

Comment:
Why is a `Call` embedded as a pointer? This can cause issues like `CallRequest.Path` panicing because `Call` is `nil`, so I want to make sure we do that for a good reason.

Reply:
@hanzei If we go the route of embedding a `Call` struct instead of a pointer, each function constructing a `CallRequest` needs to dereference the pointer to the `Call` it is embedding. That doesn't change much about the nil pointer issue, but I do agree that there's no good reason for it to be a pointer.

We do get the benefit of having a `json.Unmarshal`'d `CallRequest` avoiding this nil pointer issue. The `Path` would be empty in that case, which may cause other problems.

I'll change it to embed the struct and not the pointer

Reply:
3/5 @hanzei is right, we should embed the struct, not a pointer